core/src/main/scala/cats/data/IndexedStateT.scala

```diff
@@ -223,6 +223,11 @@ object IndexedStateT extends IndexedStateTInstances with CommonStateTConstructor
     IndexedStateT(_ => F.map(fsb)(s => (s, ())))
 }
 
+private[data] trait CommonStateTConstructors0 extends CommonStateTConstructors {
+  def empty[F[_], S, A: Monoid](implicit F: Applicative[F]): IndexedStateT[F, S, S, A] =
+    pure(Monoid[A].empty)
```

Comment:
Is there a reason to use `: Monoid` for `A` but not use `: Applicative` for `F`? I'd be inclined to not use the syntactic sugar for either so you can avoid do something like `F.pure(A.empty)` and avoid an extra `Monoid.apply[A]` call.